app/controllers/application_controller.rb

```diff
@@ -1374,6 +1374,11 @@ def find_checked_items(prefix = nil)
     end
   end
 
+  def find_checked_items_with_rbac(db, prefix = nil)
+    items = find_checked_items(prefix)
+    items.select { |item| find_by_id_filtered(db, item) }
```

Comment:
@martinpovolny says there has to be a better way

@lpichler advises low level `Rbac.filtered(items, :class => db)`

myself noticed there is a helper method `assert_rbac` for this in `ApplicationController` that might be also worth looking at.

The nice thing about `assert_rbac` is that you will get an exception when something goes wrong. Everybody knows that an exception is worth a thousand words.